core/ArchiveProcessor/Loader.php

```diff
@@ -596,12 +609,36 @@ public static function getSitesNotUsingTracker()
         return $idSitesNotUsingTracker;
     }
 
-    private function hasSiteVisitsBetweenTimeframe($idSite, Period $period)
+    private function hasSiteVisitsBetweenTimeframe($idSite, Period $period): bool
     {
         $timezone = Site::getTimezoneFor($idSite);
-        list($date1, $date2) = $period->getBoundsInTimezone($timezone);
+        /** @var \Piwik\Date $date1 */
```

Comment:
```suggestion
        /** @var Date $date1 */
        /** @var Date $date2 */
```

To have both of them typed?

Though I would also move the bound fetching past the cache check. If the cache is hit we don't need those values after all :thinking:

Reply:
Check moved, type hints added.